canal/sync.go

```diff
@@ -3,26 +3,44 @@ package canal
 import (
 	"regexp"
 	"time"
+	"fmt"
 
 	"github.com/juju/errors"
 	"github.com/ngaut/log"
 	"github.com/siddontang/go-mysql/mysql"
 	"github.com/siddontang/go-mysql/replication"
 	"github.com/siddontang/go-mysql/schema"
+	"github.com/satori/go.uuid"
 )
 
 var (
 	expAlterTable = regexp.MustCompile("(?i)^ALTER\\sTABLE\\s.*?`{0,1}(.*?)`{0,1}\\.{0,1}`{0,1}([^`\\.]+?)`{0,1}\\s.*")
 )
 
-func (c *Canal) startSyncBinlog() error {
-	pos := c.master.Position()
+func getSyncer(c *Canal) (*replication.BinlogStreamer, error) {
```

Comment:
I think that rename `startSyncBinlog` to `runSyncBinlog` will cause compatibility issues. 

What about delete `getSyncer` function and implement function logic in `startSyncBinlog`?